src/Microsoft.Azure.EventHubs.Processor/EventHubPartitionPump.cs

```diff
@@ -140,14 +140,36 @@ public PartitionReceiveHandler(EventHubPartitionPump eventHubPartitionPump)
 
             public int MaxBatchSize { get; set; }
 
-            public Task ProcessEventsAsync(IEnumerable<EventData> events)
+            public async Task ProcessEventsAsync(IEnumerable<EventData> events)
```

Comment:
Can you remove the plugin from processor? I don't really see a use case here. We can add it later if there is any ask from customers.

Reply:
Sure for now I'm only using for Sender but After Receive then won't be used shall I remove this right ? I added that because this https://github.com/Azure/azure-event-hubs-dotnet/pull/324#discussion_r223779579

Reply:
Actually that comment is for AfterMessageSend, not receive. Let's do BeforeMessageSend for now and see how that rolls out.

Reply:
Perfect yes I don't think an scenario to use `AfterMessageSend`, for Service bus probably yes because ServiceBus write for you MessageId property if you didn't.

Reply:
Fair enough. let's keep it as BeforeSend for now.